contribute/code.rst

```diff
@@ -0,0 +1,211 @@
+Code changes
+============
+
+This page helps you get your changes to Ubuntu Touch's code merged. To help you with this, we describe the things we look for in code, commits, and changesets. We also take a look at what to expect in the code review process.
+
+Assumptions
+-----------
+
+This page does not include information about how to develop applications or changes to Ubuntu Touch. To learn how to develop Ubuntu Touch components, see the :doc:`/systemdev/index` section. To learn how to develop applications, see the :doc:`/appdev/index` or :doc:`/contribute/preinstalled-apps` sections.
+
+This page assumes you know how to write your changes, use Git to commit them, push them to GitHub or Gitlab as needed, fork a repository, and make a Pull Request (PR) on GitHub or Merge Request (MR) on GitLab. We are only covering what you need to put in your changes, commit messages, and changeset descriptions; and how to expect the review process will work.
+
+Definitions
+-----------
+
+Since we use GitHub and GitLab for our code hosting depending on the component, some terms must be defined up front for both platforms.
+
+changeset
+    A GitHub Pull Request (PR) or GitLab Merge Request (MR).
+issue report
+    A GitHub or GitLab Issue.
+
+Reasoning
+---------
+
+Ubuntu Touch is used by many users on many different devices. Breaking functionality that these users rely on can be catastrophic. Our review process helps us avoid broken functionality by ensuring a few properties of the changeset.
+
+In particular, we ask: Does the change...
+
+* meet its own goals, such as adding a feature or fixing a bug?
+* cause any unexpected behavior?
+* make it easy for someone to make further changes to the software in the future?
+
```

Comment:
I am not 100% sure if there are meanings of "Reasoning" I don't know. So I might be wrong with this one. But I think questions are not "giving reasons".

If kept as questions, I think it ("In particular..." up to "...in the future?") would be better placed somewhere under commit description maybe.

Or rephrase to aims since we want to give reasons:

In particular we want to make sure the change does...
* meets its own goals, such as adding a feature or fixing a bug
* does not cause any unexpected behavior
* makes it easy for someone to make further changes to the software in the future